CRM/Contribute/Selector/Search.php

```diff
@@ -402,6 +402,13 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
         }
       }
 
+      $row['balance_due'] = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(
+        $result->contribution_id,
```

Comment:
Hi - if you have agreed to go ahead with @colemanw then I guess he'll have to honour that - please do it as a setting that is opt-out by default and then no-one else will be affected.

If you DO impose this on all sites then you effectively transfer the work onto me of making it not run extra queries before I can upgrade.